### PR TITLE
feat: add support for node v18

### PR DESCRIPTION
**Pull request by frbuceta**
Signed-off-by: Francisco Buceta <frbuceta@gmail.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node-version: [12, 14, 16, 17]
+        node-version: [12, 14, 16, 18]
```
Comment (achrinza) [inline .github/workflows/continuous-integration.yml:23]: Node.js v17 is still supported, hence we should continue testing it until it EOLs in June 2022. It isn't  a long time from now, but still not EOL yet nonetheless.

```suggestion
        node-version: [12, 14, 16, 17, 18]
```

File: acceptance/extension-logging-fluentd/package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "directory": "acceptance/extension-logging-fluentd"
   },
   "engines": {
-    "node": "12 || 14 || 16 || 17"
+    "node": "12 || 14 || 16 || >=18"
```
Comment (achrinza) [inline acceptance/extension-logging-fluentd/package.json:17]: We have been conservative with the engine version range as it's used as a signal for "officially tested & support Node.js versions"

```suggestion
    "node": "12 || 14 || 16 || 17 || 18"
```

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -34,8 +34,6 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Update NPM
-        run: npm install -g npm
```
Comment (achrinza) [inline .github/workflows/continuous-integration.yml:38]: This causes the same issue as https://github.com/loopbackio/loopback-connector-postgresql/pull/499#discussion_r853157977.

**Review by achrinza — CHANGES_REQUESTED** [review]
`README.md` needs to be updated as well.

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (frbuceta) [reply .github/workflows/continuous-integration.yml:23]: But v17 is not an LTS version and never will be. Should we support him? These versions are discouraged for production.

File: .github/workflows/continuous-integration.yml (hunk 3 above)
Comment (frbuceta) [reply .github/workflows/continuous-integration.yml:38]: Solved

File: README.md [hunk 4]
```diff
@@ -56,7 +56,9 @@ for more details.
 
 Make sure you have the following installed:
 
-- [Node.js](https://nodejs.org/en/download/) >= 12
+| Package                                    | Version/-s                                                                             | Link                                  | Note                                                                                                  |
```
Comment (achrinza) [inline README.md:59]: The EOL for LB4 in the LTS matrix a few lines above needs to be updated as well:

```diff
  | Version    | Status      | Published | EOL                  |
  | ---------- | ----------- | --------- | -------------------- |
- | LoopBack 4 | Current     | Oct 2018  | Apr 2024 _(minimum)_ |
+ | LoopBack 4 | Current     | Oct 2018  | Apr 2025 _(minimum)_ |
  | LoopBack 3 | End-of-Life | Dec 2016  | Dec 2020             |
  | LoopBack 2 | End-of-Life | Jul 2014  | Apr 2019             |
```

**Review by achrinza — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (frbuceta) [reply README.md:59]: Solve

**Pull request merged**